scripts/xlinear_predict/predict.py

```diff
@@ -0,0 +1,311 @@
+            only_topk=self.only_topk,
+            threshold=self.min_weight_value,
+        )
+        return self
+
+    def predict(self, X):
+        return self.predict_proba(X) > self.threshold
+
+    def predict_proba(self, X):
+        if self.vectorizer_library == "sklearn":
+            return self.xlinear_model_.predict(
+                self.vectorizer_.transform(X).astype("float32"),
+                only_topk=self.only_topk,
+                beam_size=self.beam_size,
+            )
+        else:
+            return self.xlinear_model_.predict(
+                self.vectorizer_.predict(X).astype("float32"),
+                only_topk=self.only_topk,
+                beam_size=self.beam_size,
+            )
+
+    def save(self, model_path):
+        model_path = str(model_path)  # In case a Posix is passed
+        params_path = os.path.join(model_path, "params.json")
+        vectorizer_path = os.path.join(model_path, "vectorizer.pkl")
+        with open(params_path, "w") as f:
+            json.dump(self.__dict__, f, indent=4, default=str)
+
+        if self.vectorizer_library == "sklearn":
+            save_pickle(vectorizer_path, self.vectorizer_)
+        else:
+            self.vectorizer_.save(model_path)
+
+        self.xlinear_model_.save(model_path)
+
+    def load(self, model_path, is_predict_only=True):
+        model_path = str(model_path)  # In case a Posix is passed
+        params_path = os.path.join(model_path, "params.json")
+        vectorizer_path = os.path.join(model_path, "vectorizer.pkl")
+        with open(params_path, "r") as f:
+            self.__dict__.update(json.load(f))
+
+        if self.vectorizer_library == "sklearn":
+            self.vectorizer_ = load_pickle(vectorizer_path)
+        else:
+            from pecos.utils.featurization.text.vectorizers import Tfidf
+
+            self.vectorizer_ = Tfidf()
+            self.vectorizer_ = self.vectorizer_.load(model_path)
+
+        self.xlinear_model_ = XLinearModel.load(
+            model_path, is_predict_only=is_predict_only
+        )
+
+
+def create_model(parameters=None):
+    """Creates an XLinear model
+
+    Args:
+    - parameters: parameters to set the XLinear model
+
+    Returns:
+        an xlinear model
+
+    """
+    model = MeshXLinear()
+
+    if parameters:
+        params = ast.literal_eval(parameters)
+        model.set_params(**params)
+    else:
+        parameters = {}
+    return model
+
+
+def load_model(model_path, parameters=None):
+    """Loads an XLinear model from a pickle file
+
+    Args:
+    - model_path: location of pickled model
+    - parameters: parameters to set the XLinear model
+
+    Returns:
+        an xlinear model
+
+    """
+    if str(model_path).endswith(".pkl"):
+        model = load_pickle(model_path)
+    else:
+        model = create_model(parameters=parameters)
+        model.load(model_path)
+
+    return model
+
+
+def predict_tags(
```

Comment:
no need for most other functions like `save_model`, `load_model`, `create_model`, `format_predictions` just have one function that calls the model and formats the predictions in the same way as bertmesh